pkg/cmd/clients/factory.go

```diff
@@ -386,6 +386,17 @@ func (f *factory) CreateAuthConfigService(fileName string, namespace string,
 	return nil, fmt.Errorf("no auth config found for secret %q", fileName)
 }
 
+// CreateLocalGitAuthConfigService creates a new service which loads/saves the auth config from/to a local file.
+func (f *factory) CreateLocalGitAuthConfigService() (auth.ConfigService, error) {
```

Comment:
Me and @ccojocar have recently been talking about git and the various authentication needs around it. Personally I am not a fan using `gitAuth.yaml` and writing the credentials to disk. 

I was more envisioning a custom git credential helper or the use of `GIT_ASKPASS` and the ability to call into `jx` itself to get the appropriate credentials. Here is a PR I created a while back using `GIT_ASKPASS` - https://github.com/jenkins-x/jx/pull/5850/. We did not merge this since @ccojocar was working on his auth refactoring. I am pretty sure `GIT_ASKPASS` would work in this scenario as well w/o the need to write anything to disk.